include/DirectSolver.hpp

```diff
@@ -11,14 +11,15 @@
 #include "Types.hpp"
 
 namespace blitzdg {
-  class DirectSolver {
-      index_type N;
-      SparseMatrixConverter MatrixConverter;
+    class DirectSolver {
+        SparseMatrixConverter MatrixConverter;
   
-  public:
-      DirectSolver(SparseMatrixConverter const &);
+    public:
+        DirectSolver()
+        : MatrixConverter{}
+        {}
 
-      void solve(const matrix_type& A, const matrix_type& B, matrix_type& X);
+        void solve(const matrix_type& A, const matrix_type& B, matrix_type& X) const;
```

Comment:
what does adding const after the signature here do? Is it telling the compiler that this method is not allow to modify class members?

Reply:
Yes, const after a member function name means that member function cannot modify the object on which it is called.